src/core/dyad_core.c

```diff
@@ -0,0 +1,781 @@
+    // for a producer operation. So, return DYAD_BADMANAGEDPATH
+    if (check_prod && (ctx->prod_managed_path == NULL || strlen(ctx->prod_managed_path) == 0))
+    {
+        return DYAD_BADMANAGEDPATH;
+    }
+    // If none of the above conditions were met, then the context is valid for the desired
+    // operation. So, return DYAD_OK
+    return DYAD_OK;
+}
+
+static int gen_path_key (const char* str, char* path_key, const size_t len,
+        const uint32_t depth, const uint32_t width)
+{
+    static const uint32_t seeds [10]
+        = {104677u, 104681u, 104683u, 104693u, 104701u,
+            104707u, 104711u, 104717u, 104723u, 104729u};
+
+    uint32_t seed = 57u;
+    uint32_t hash[4] = {0u}; // Output for the hash
+    size_t cx = 0ul;
+    int n = 0;
+
+    if (path_key == NULL || len == 0ul) {
+        return -1;
+    }
+    path_key [0] = '\0';
+
+    for (uint32_t d = 0u; d < depth; d++)
+    {
+        seed += seeds [d % 10];
+        // TODO add assert that str is not NULL
+        MurmurHash3_x64_128 (str, strlen (str), seed, hash);
+        uint32_t bin = (hash[0] ^ hash[1] ^ hash[2] ^ hash[3]) % width;
+        n = snprintf (path_key+cx, len-cx, "%x.", bin);
+        cx += n;
+        if (cx >= len || n < 0) {
+            return -1;
+        }
+    }
+    n = snprintf (path_key+cx, len-cx, "%s", str);
+    if (cx + n >= len || n < 0) {
+        return -1;
+    }
+    return 0;
+}
+
+#if DYAD_PERFFLOW
+__attribute__((annotate("@critical_path()")))
+static int dyad_kvs_commit(dyad_ctx_t *ctx, flux_kvs_txn_t *txn,
+        const char *upath, flux_future_t **f)
+#else
+static inline int dyad_kvs_commit(dyad_ctx_t *ctx, flux_kvs_txn_t *txn,
+        const char *upath, flux_future_t **f)
+#endif
+{
+    // Commit the transaction to the Flux KVS
+    printf("Committing data to KVS\n");
+    *f = flux_kvs_commit(ctx->h, ctx->kvs_namespace,
+            0, txn);
+    // If the commit failed, log an error and return DYAD_BADCOMMIT
+    if (*f == NULL)
+    {
+        DYAD_LOG_ERR(ctx, "Could not commit transaction to Flux KVS\n");
+        return DYAD_BADCOMMIT;
+    }
+    // If the commit is pending, wait for it to complete
+    printf("Wait for commit to complete\n");
+    flux_future_wait_for(*f, -1.0);
+    printf("Destroy future for KVS commit\n");
+    // Once the commit is complete, destroy the future and transaction
+    flux_future_destroy(*f);
+    *f = NULL;
+    printf("Destroy transaction on KVS\n");
+    flux_kvs_txn_destroy(txn);
+    return DYAD_OK;
+}
+
+#if DYAD_PERFFLOW
+__attribute__((annotate("@critical_path()")))
+static int publish_via_flux(dyad_ctx_t* ctx, const char *upath)
+#else
+static inline int publish_via_flux(dyad_ctx_t* ctx, const char *upath)
+#endif
+{
+    printf("Initial publication setup\n");
+    int ret = 0;
+    const char* prod_managed_path = NULL;
+    flux_kvs_txn_t *txn = NULL;
+    flux_future_t *f = NULL;
+    const size_t topic_len = PATH_MAX;
+    char topic[topic_len+1] = {'\0'};
```

Comment:
`topic[topic_len+1]` is problematic with some C compilers.
It should be `topic[PATH_MAX+1]`

Reply:
Resolved